jaxley/modules/branch.py

```diff
@@ -146,6 +148,51 @@ def _init_morph_jax_spsolve(self):
     def __len__(self) -> int:
         return self.nseg
 
+    def set_ncomp(self, ncomp: int, min_radius: Optional[float] = None):
+        """Set the number of compartments with which the branch is discretized.
+
+        Args:
+            ncomp: The number of compartments that the branch should be discretized
+                into.
+
+        Raises:
+            - When the Module is a Network.
+            - When there are stimuli in any compartment in the Module.
+            - When there are recordings in any compartment in the Module.
+            - When the channels of the compartments are not the same within the branch
+            that is modified.
+            - When the lengths of the compartments are not the same within the branch
+            that is modified.
+            - Unless the morphology was read from an SWC file, when the radiuses of the
+            compartments are not the same within the branch that is modified.
+        """
+        assert len(self.externals) == 0, "No stimuli allowed!"
+        assert len(self.recordings) == 0, "No recordings allowed!"
+        assert len(self.trainable_params) == 0, "No trainables allowed!"
+
+        # Update all attributes that are affected by compartment structure.
+        (
+            self.nodes,
+            self.nseg_per_branch,
+            self.nseg,
+            self.cumsum_nseg,
+            self._internal_node_inds,
+        ) = self._set_ncomp(
+            ncomp,
+            self.nodes,
+            self.nodes,
+            self.nodes["comp_index"].to_numpy()[0],
+            self.nseg_per_branch,
+            [c._name for c in self.channels],
+            list(chain(*[c.channel_params for c in self.channels])),
+            list(chain(*[c.channel_states for c in self.channels])),
+            self._radius_generating_fns,
+            min_radius,
+        )
+
+        # Update the morphology indexing (e.g., `.comp_edges`).
+        self.initialize()
+
 
 class BranchView(View):
```

Comment:
does branchview have a method / arg to check the number of segments? 
I think it would be nice if `cell.branch(0).nseg` or similar would show the correct number.

Reply:
I am not sure if we need this---`cell.branch(0).show()` will have `nseg` rows, I think it would be fine for the user to just do this (as compared to having another magical argument which the user has to learn how to use).